ElectronNET.API/App.cs

```diff
@@ -464,40 +464,47 @@ public void Show()
         /// The current application directory.
         /// </summary>
         /// <returns></returns>
-        public async Task<string> GetAppPathAsync()
+        public async Task<string> GetAppPathAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
-            var taskCompletionSource = new TaskCompletionSource<string>();
+            cancellationToken.ThrowIfCancellationRequested();
 
-            BridgeConnector.Socket.On("appGetAppPathCompleted", (path) =>
+            var taskCompletionSource = new TaskCompletionSource<string>();
+            using(cancellationToken.Register(() => taskCompletionSource.TrySetCanceled()))
```

Comment:
I deliberately choose here the non state based overload. This leads to closure allocations but since the BridgeConnector has the same problem I figured it is OK